File: backend spring/src/main/java/com/bloodbank/dto/response/DonorInfoResponseDto.java
package com.bloodbank.dto.response;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Date;

@Getter
@Setter
public class DonorInfoResponseDto {
    private Long userId;
    private String email;
    private String fullName;
    private String bloodGroup;
    private String contactNumber;
    private LocalDate lastDonationDate; // Change the type here
    private boolean isEligible;
    private LocalDateTime createdAt;
}

File: BackendDotNet/DTOs/Response/DonorInfoResponseDto.cs
namespace BackendDotNet.DTOs.Response
{
    public class DonorInfoResponseDto
    {
        public long UserId { get; set; }
        public string Email { get; set; } = string.Empty;
        public string FullName { get; set; } = string.Empty;
        public string BloodGroup { get; set; } = string.Empty;
        public string ContactNumber { get; set; } = string.Empty;
        public DateTime? LastDonationDate { get; set; }
        public bool IsEligible { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}
